qutip/tests/core/test_metrics.py

```diff
@@ -453,6 +453,15 @@ def test_force_solve(self, dimension, generator):
             == pytest.approx(dnorm(A, B, force_solve=True), abs=1e-5)
         )
 
+    @pytest.mark.repeat(3)
+    def test_unitary_case(self, dimension):
+        """Check that the diamond norm is one for unitary maps."""
+        A, B = rand_unitary(dimension), rand_unitary(dimension)
+        assert (
+            dnorm(A, B)
+            == pytest.approx(dnorm(A, B, force_solve=True), abs=1e-5)
+        )
+
```

Comment:
This is redundant to `test_force_solve`...

Reply:
Sorry I missed this. Perhaps `force_solve=True` should be left out?

Reply:
I think the unitary case is already tested enough and no new tests are needed.

Reply:
I agree to some extent, but it would be nice to have a test explicitly for this case since now it is treated with a separate code path.

Reply:
Regarding this, I think having a unitary test is probably a good idea since we now do have an explicit optimisation for a difference of unitaries. Regarding `test_force_solve`, I also think it's redundant. To be honest I think we should just get rid of `test_force_solve` as it doesn't really accomplish anything. For the `rand_unitary` generator it's exactly the same as the new `test_unitary_case`. For the `rand_super_bcsz` generator, I'm sort of confused what's being tested in the first place. There is not optimisation for a difference of CPTP channels, the semidefinite program is run whether `test_force_solve` is set to `True` or not (the difference of CPTP channels is not completely positive in general.)

Reply:
I think we ought to keep `test_unitary_case` and get rid of `test_force_solve`. Perhaps it's also worth writing a test for the completely positive channel optimisation.

Reply:
+1 on adding a `test_completely_positive_case` or `test_cp_case` and `test_cptp_case`.

I didn't look at `test_force_solve` but I expect it to pass `force_solve=True` while I think `test_unitary_case` should pass `force_solve=False`.

@Ericgig Thoughts?

Reply:
Great, `test_cptp` already exists, although renaming to `test_cptp_case` for consistency is maybe worthwhile. 

I need to figure out how best to implement `test_cp_case`, any suggestions would be great.

`test_force_solve` actually just compares the result of `dnorm` when  `force_solve` is set to `True` and when it's set to `False`. All `force_solve` does is force `dnorm` to solve the semidefinite program rather than choose an optimisation if it's available, so it's a good baseline to test against.

As things stand, in `test_unitary_case` I compare the result of `dnorm` with unitary difference optimisation to the baseline semidefinite program (`force_solve = True`). I think this is the correct way to run the test in the absence of any analytical formulas.

Reply:
I am for removing `test_force_solve` and splitting each case in it's own test.
Using `force_solve=True` in `test_unitary_case` and other tests to ensure that the optimised algorithm give the same result as the normal one is the way to go.

Reply:
Have committed these changes @Ericgig. Let me know if it is worth having a `test_force_solve` case regardless.